### ISSUE TITLE
GH actions on pushes

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#3040]
Hi @LorenzoBettini 

i see no build started for `[cd/jetty11021](https://github.com/eclipse/xtext/tree/cd/jetty11021)` any idea?

**Comment by cdietrich** [comment]
no i dont know anything. @eclipsewebmaster do you have an idea?

**Comment by cdietrich** [comment]
created https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4640

**Comment by cdietrich** [comment]
In ci hitting scan repository now usually works

**Comment by LorenzoBettini** [comment]
Yes, but I meant I don't see the PR build on JIRO... maybe there we do the same on JIRO, not building internal PRs?

Anyway, the PR doesn't show the link to the build on JIRO.

**Comment by cdietrich** [comment]
Yes think so

**Comment by fredg02** [comment]
> i see no build started for [cd/jetty11021](https://github.com/eclipse/xtext/tree/cd/jetty11021) any idea?

I see two attempts that were skipped:
* https://github.com/eclipse/xtext/actions/runs/9109823886/attempts/1
* https://github.com/eclipse/xtext/actions/runs/9109823886/attempts/2

Not sure why they were skipped.

> Or maybe it's some problem introduced by the Eclipse Foundation?

I'm not aware of any changes from our side that could have caused GH actions to be skipped.

FTR: There was an incident with GH actions recently: https://www.githubstatus.com/incidents/2bglpcjvg7xb

> I don't know if it's related or simply delayed, but the PR build does not start on JIRO either https://github.com/eclipse/xtext/pull/3041 for you

Jenkins Multibranch pipeline scan log shows:
```
Checking pull request [#3041](https://github.com/eclipse/xtext/pull/3041)
    Submitted from origin repository, skipping
```

**Comment by fredg02** [comment]
Side note: the xtext GH repo has not been using a webhook to trigger builds on their EF Jenkins instance. Instead, it relied on polling for changes every 2 minutes. 

I have set up a new webhook that triggers builds on pushes and pull requests. The polling frequency has been adjusted to once per day, if not otherwise run (e.g. by the webhook trigger).

@cdietrich there are webhooks that don't seem to work any more (xtext-builds.itemis.de and hooks.slack.com). Can they be removed?

**Comment by LorenzoBettini** [comment]
@fredg02 some more information: PR jobs are intentionally skipped for internal PRs https://github.com/eclipse/xtext/blob/main/.github/workflows/maven.yml#L14

The problem is that for @cdietrich standard jobs for pushes are simply not started (they're not even skipped).

**Comment by fredg02** [comment]
> @fredg02 some more information: PR jobs are intentionally skipped for internal PRs https://github.com/eclipse/xtext/blob/main/.github/workflows/maven.yml#L14

Ok. That explains the behavior for PRs then.

> The problem is that for @cdietrich standard jobs for pushes are simply not started (they're not even skipped).

I see that actions ran for recent commits from @cdietrich here https://github.com/eclipse/xtext/commits/cd-jetty11021 and https://github.com/eclipse/xtext/commits/main. Can you point me to a commit where actions were not started?

**Comment by LorenzoBettini** [comment]
> > @fredg02 some more information: PR jobs are intentionally skipped for internal PRs https://github.com/eclipse/xtext/blob/main/.github/workflows/maven.yml#L14
> 
> Ok. That explains the behavior for PRs then.
> 
> > The problem is that for @cdietrich standard jobs for pushes are simply not started (they're not even skipped).
> 
> I see that actions ran for recent commits from @cdietrich here https://github.com/eclipse/xtext/commits/cd-jetty11021 and https://github.com/eclipse/xtext/commits/main. Can you point me to a commit where actions were not started?

@fredg02 I don't see any actions for recent @cdietrich commits you mentioned above. I only see the intended skipped jobs for PR, but I don't see the actions for pushes:

![image](https://github.com/eclipse/xtext/assets/1202254/902cb9c0-61cf-489e-ae37-6cd4dc7aa33e)

This is how it should look like

![image](https://github.com/eclipse/xtext/assets/1202254/a84a03ea-d04a-4181-8c8f-59fda77b6f7f)

note the skipped jobs for PR, but the executed jobs for pushes.

What I'm saying is that jobs for pushes just don't appear (they're not skipped)

**Comment by cdietrich** [comment]
just pushed `cd-xyz`without a pr and nothing starts

**Comment by fredg02** [comment]
@LorenzoBettini 

It worked for at least the merge commit here:
![image](https://github.com/eclipse/xtext/assets/605957/ac350769-d02e-4a53-8a6e-e69b8737b677)

For the other commits in the cd-jetty11021 branch, where one can still see the checks, the GH action attributes the commits to PRs and skips it.

**Comment by cdietrich** [comment]
and for `cd-xyz`?

**Comment by fredg02** [comment]
> just pushed cd-xyzwithout a pr and nothing starts

@cdietrich "nothing starts" is not what I see here:
![image](https://github.com/eclipse/xtext/assets/605957/c3871e14-032b-46cc-813b-b6c546c1345f)

CI build on Jenkins is pending, and GH actions are skipped since (again) they are (for some reason) recognized as pull request.

Instead of creating a new branch, can you commit and push a simple change to `cd-xyz`, e.g. add a whitespace in a readme file or something else that is not a code change?

**Comment by cdietrich** [comment]
did a dummy change

**Comment by cdietrich** [comment]
how are the "Workflow permissions" configured in the repos settings?

**Comment by fredg02** [comment]
> if you look at the "Update to LSP4J 0.23.0. Fixes..." commit above, you'll see that build jobs were not started.

The build jobs were skipped, the checks say `Build / build (pull_request) Skipped`. This is different from "build jobs were not started".

![image](https://github.com/eclipse/xtext/assets/605957/3b8a52e6-1882-4cd0-8eb2-7d66df572382)

What you were talking about, and I did not see and asking about was such a dummy commit (in the screenshot) where NOTHING (no checks) are running.

And at least for this case, there is a good reason here:
https://github.com/eclipse/xtext/blob/main/.github/workflows/maven.yml#L6-L7

I had not noticed the `path-ignores ` until after I asked @cdietrich to commit a change to a readme file. @cdietrich can you commit a dummy change that is not in a *.md file, please?

**Comment by cdietrich** [comment]
ok will change something else

**Comment by fredg02** [comment]
> how are the "Workflow permissions" configured in the repos settings?

Look normal to me. Nothing that should cause different behavior for different users.

**Comment by cdietrich** [comment]
but in the jetty case i also changed other stuff.

**Comment by cdietrich** [comment]
hmmmm
kinda redid the jetty change and now it starts too.

**Comment by cdietrich** [comment]
maybe there was something wrong with the polling

**Comment by LorenzoBettini** [comment]
Or the mentioned recent problem wasn't yet completely fixed...

**Comment by LorenzoBettini** [comment]
> Side note: the xtext GH repo has not been using a webhook to trigger builds on their EF Jenkins instance. Instead, it relied on polling for changes every 2 minutes.
> 
> I have set up a new webhook that triggers builds on pushes and pull requests. The polling frequency has been adjusted to once per day, if not otherwise run (e.g. by the webhook trigger).
> 

@fredg02 not related to this issue, but I guess your change is the cause of this new issue https://github.com/eclipse/xtext/issues/3049

currently, we are checking the build cause like that

```
currentBuild.getBuildCauses().toString().contains('BranchIndexingCause')
```

But of course, now this doesn't work anymore... could you please help us change the condition so that it is true when the event is a push?

I cannot seem to find documentation on the expected output of such obscure and untyped groovy calls...

thanks in advance

**Comment by LorenzoBettini** [comment]
> > Side note: the xtext GH repo has not been using a webhook to trigger builds on their EF Jenkins instance. Instead, it relied on polling for changes every 2 minutes.
> > I have set up a new webhook that triggers builds on pushes and pull requests. The polling frequency has been adjusted to once per day, if not otherwise run (e.g. by the webhook trigger).
> 
> @fredg02 not related to this issue, but I guess your change is the cause of this new issue #3049
> 
> currently, we are checking the build cause like that
> 
> ```
> currentBuild.getBuildCauses().toString().contains('BranchIndexingCause')
> ```
> 
> But of course, now this doesn't work anymore... could you please help us change the condition so that it is true when the event is a push?
> 
> I cannot seem to find documentation on the expected output of such obscure and untyped groovy calls...
> 
> thanks in advance

Never mind: I fixed that
